lib/ansible/modules/network/nxos/nxos_interface.py

```diff
@@ -399,339 +289,474 @@ def get_interfaces_dict(module):
+                output = run_commands(module, cmd, check_rc=False)
+                if output:
+                    have_neighbors = output[0]
+                else:
+                    have_neighbors = ''
+                if have_neighbors and 'Total entries displayed: 0' not in have_neighbors:
+                    for line in have_neighbors.strip().split('\n'):
+                        if line.startswith('Port Description'):
+                            have_port.append(line.split(': ')[1])
+                        if line.startswith('System Name'):
+                            have_host.append(line.split(': ')[1])
+
+            for item in want_neighbors:
+                host = item.get('host')
+                port = item.get('port')
+                if host and host not in have_host:
+                    failed_conditions.append('host ' + host)
+                if port and port not in have_port:
+                    failed_conditions.append('port ' + port)
+
+    return failed_conditions
+
+
 def main():
+    """ main entry point for module execution
+    """
+    neighbors_spec = dict(
+        host=dict(),
+        port=dict()
+    )
+
+    element_spec = dict(
+        name=dict(aliases=['interface']),
+        admin_state=dict(default='up', choices=['up', 'down']),
+        description=dict(),
+        speed=dict(),
+        mode=dict(choices=['layer2', 'layer3']),
+        mtu=dict(),
+        duplex=dict(choices=['full', 'half', 'auto']),
+        interface_type=dict(choices=['loopback', 'portchannel', 'svi', 'nve']),
+        ip_forward=dict(choices=['enable', 'disable']),
+        fabric_forwarding_anycast_gateway=dict(type='bool'),
+        tx_rate=dict(),
+        rx_rate=dict(),
+        neighbors=dict(type='list', elements='dict', options=neighbors_spec),
+        delay=dict(default=10, type='int'),
+        state=dict(choices=['absent', 'present', 'default'], default='present')
+    )
+
+    aggregate_spec = deepcopy(element_spec)
+    aggregate_spec['name'] = dict(required=True)
+
+    # remove default in aggregate spec, to handle common arguments
+    remove_default_spec(aggregate_spec)
 
     argument_spec = dict(
-        interface=dict(required=False,),
-        admin_state=dict(default='up', choices=['up', 'down'], required=False),
-        description=dict(required=False, default=None),
-        mode=dict(choices=['layer2', 'layer3'], required=False),
-        mtu=dict(type='int', required=False),
-        interface_type=dict(required=False, choices=['loopback', 'portchannel', 'svi', 'nve']),
-        ip_forward=dict(required=False, choices=['enable', 'disable']),
-        fabric_forwarding_anycast_gateway=dict(required=False, type='bool'),
-        state=dict(choices=['absent', 'present', 'default'], default='present', required=False)
+        aggregate=dict(type='list', elements='dict', options=aggregate_spec),
```

Comment:
Please add `mutually_exclusive = [['name', 'interface_type']]` in aggregate spec dict. This ensure the mutual exclusion is applied for each item in aggreaget list

Reply:
fixed!